### PR TITLE
F 1538 Implement onboard.js

### PR DESCRIPTION
**Pull request by RamRamez**
Issue #1538

### CODE REVIEW

File: src/services/DonationBlockchainService.jsx [hunk 1]
```diff
@@ -490,54 +471,50 @@ class DonationBlockchainService {
    * @param {string}   address   Address of the user who calls commit
    * @param {function} onCreated Callback function after the transaction has been broadcasted to chain and stored in feathers
    * @param {function} onSuccess Callback function after the transaction has been mined
-   * @param {function} onError   Callback function after error happened
+   * @param liquidPledging       liquidPledging
    */
-  static commit(donation, address, onCreated = () => {}, onSuccess = () => {}, onError = () => {}) {
+  static commit(donation, address, onCreated = () => {}, onSuccess = () => {}, liquidPledging) {
```
Comment (aminlatifi) [inline src/services/DonationBlockchainService.jsx:476]: Please pass web3 like other places in project.

File: src/contextProviders/Web3Provider.jsx [hunk 2]
```diff
@@ -293,12 +107,13 @@ class Web3Provider extends Component {
...
       isHomeNetwork,
       isForeignNetwork,
-      isEnabled,
-      setupTimeout,
       showForeignNetRequiredWarning,
       showHomeNetRequiredWarning,
+      web3,
     } = this.state;
 
+    const isEnabled = !!account && !!balance;
```
Comment (aminlatifi) [inline src/contextProviders/Web3Provider.jsx:115]: Is `&& !!balance` needed here?

**Review by aminlatifi — CHANGES_REQUESTED** [review]
Thanks @RamRamez big improvement

File: src/contextProviders/Web3Provider.jsx (hunk 2 above)
Comment (RamRamez) [reply src/contextProviders/Web3Provider.jsx:115]: We have four subscriptions: web3, network, address and balance. First web3 become ready and last balance will be ready. On each of these four, App will be rerendered by Web3 Context Provider. I think `!!balance` is enough but to make sure we can include all four subscriptions in that conditions.

**Pull request merged**